DOC-550: move .NET client tech content into docs (#1805)
hazelcast/hz-docs: docs/modules/clients/pages/dotnet.adoc

:page-api-reference: http://hazelcast.github.io/hazelcast-csharp-client/{page-latest-supported-csharp-client}/api/index.html
[[net-client]]

> TIP: For the latest .NET API documentation, see http://hazelcast.github.io/hazelcast-csharp-client/{page-latest-supported-csharp-client}/api/index.html[Hazelcast .NET Client docs].

Reviewer: Just checking it's deliberate to omit the explicit API docs link?

Author: I'll include it in the Next steps section, although it's accessible from the Docs link above

This section provides information about the .NET client for Hazelcast, and explains how to install and get started with the client. 

> TIP: To learn how to get started quickly with the Hazelcast .NET client, follow our simple xref:clients:csharp-client-getting-started.adoc[Get started with .NET] tutorial.  

Reviewer: ```suggestion
TIP: To learn how to get started quickly with the Hazelcast .NET client, follow the xref:clients:csharp-client-getting-started.adoc[Get started with .NET] tutorial.  
```

Author: done

The official Hazelcast .NET client allows .NET applications to connect to and interact with Hazelcast clusters. 
The key features and benefits include:

> * Distributed Data Structures: it provides access to Hazelcast's distributed data structures such as maps, queues, topics, and more.
> * SQL Support: the client allows running SQL queries over distributed data.
> * Near Cache: it supports Near Cache for faster read speeds. Eventual consistency is also supported.
> * Security Features: the client offers SSL support and Mutual Authentication for enterprise-level security needs.
> * JSON Object Support: it allows using and querying JSON objects.
> * Zero Downtime Upgrades: Blue/Green failover for zero downtime during upgrades is supported.
> * Scalability: the Hazelcast .NET Client is designed to handle increasing workloads efficiently, making it suitable for large-scale applications.

Reviewer: ```suggestion
* Distributed data structures: the client provides access to Hazelcast's distributed data structures such as maps, queues, topics, and more.
* SQL support: the client enables you to run SQL queries over distributed data.
* Near cache: the client supports near cache for faster read speeds. Eventual consistency is also supported.
* Security features: the client offers SSL support and mutual authentication for enterprise-level security needs.
* JSON object support: enables you to use and query JSON objects.
* Zero downtime upgrades: the client supports blue/green failover for zero downtime during upgrades.
* Scalability: the Hazelcast .NET Client is designed to handle increasing workloads efficiently, making it suitable for large-scale applications.
```
Some capitalization suggestions and more explicit mentions of client to support LLM parsing

These features make the Hazelcast .NET Client ideal for .NET applications requiring high-performance, scalable, and distributed data processing capabilities.

TIP: For the latest .NET API documentation, see http://hazelcast.github.io/hazelcast-csharp-client/{page-latest-supported-csharp-client}/api/index.html[Hazelcast .NET Client docs].

== Install the .NET client

This section explains how to download and install the Hazelcast .NET client using the published Hazelcast.Net package from NuGet. 

=== Requirements

The Hazelcast .NET client is distributed as a NuGet package that targets .NET Standard versions .NET 6 and .NET 8. It can therefore be used in any application targetting .NET versions that support these .NET Standard versions:

* .NET Framework 4.6.2 and 4.8, on Windows
* .NET Core, on Windows and Linux

> For details about the exact versions supported by the client, see http://hazelcast.github.io/hazelcast-csharp-client/versions.html[versions]. 

Reviewer: ```suggestion
For details about the exact versions supported by the client, see http://hazelcast.github.io/hazelcast-csharp-client/versions.html[Versions]. 
```

=== Distribution

The .NET client is distributed via NuGet as a package named `Hazelcast.Net``. It can be installed like any other NuGet package, either via the Visual Studio GUI, or using the package manager:

```
PM> Install-Package Hazelcast.Net
```
Or via the .NET CLI:

```
> dotnet add package Hazelcast.Net
```
Or manually added to the project as a package reference:

```
<PackageReference Include="Hazelcast.Net" Version="4.0.0" />
```

> NOTE: When including the Hazelcast.Net package in a .NET Framework project, note that some binding redirects may be required. For information, see http://hazelcast.github.io/hazelcast-csharp-client/{page-latest-supported-csharp-client}/doc/download-install.html#binding-redirects[Binding Redirects].

Reviewer: ```suggestion
NOTE: When including the Hazelcast.Net package in a .NET Framework project, note that some binding redirects may be required. For more information, see http://hazelcast.github.io/hazelcast-csharp-client/{page-latest-supported-csharp-client}/doc/download-install.html#binding-redirects[Binding Redirects].
```

**1. Create a .NET console project and add the Hazelcast client:**

```sh
   dotnet new console -n hazelcast-example
   cd hazelcast-example
   dotnet add package Hazelcast.Net
```

**2. Edit your `Program.cs` file with the following code:**

```csharp
   using System.Threading.Tasks;
   using Hazelcast;

   class Program
   {
       public static async Task Main()
       {
           // Create default options and connect to a Hazelcast server running on localhost
           var options = new HazelcastOptionsBuilder().Build();
           await using var client = await HazelcastClientFactory.StartNewClientAsync(options);

           // Use the client (e.g., get a distributed map)
           var map = await client.GetMapAsync<string, string>("my-distributed-map");
           await map.SetAsync("key", "value");
           var value = await map.GetAsync("key");
           System.Console.WriteLine($"Value for 'key': {value}");

           // Dispose the client when done
           await client.DisposeAsync();
       }
   }
```
**3. Run your application:**

```sh
   dotnet run
```

> This will connect your .NET application to a Hazelcast cluster and allow you to interact with distributed data structures like maps. For more advanced configuration (e.g., connecting to Hazelcast Cloud, using SSL, or customizing logging), refer to the https://hazelcast.github.io/hazelcast-csharp-client/latest/doc/getting-started.html#using-the-client[Getting Started documentation] and the https://github.com/hazelcast/hazelcast-csharp-client/tree/master/src/Hazelcast.Net.Examples[code samples]. 

Reviewer: ```suggestion
This will connect your .NET application to a Hazelcast cluster and allow you to interact with distributed data structures like maps. For more advanced configuration (e.g. connecting to Hazelcast Cloud, using SSL, or customizing logging), refer to the https://hazelcast.github.io/hazelcast-csharp-client/latest/doc/getting-started.html#using-the-client[Getting Started documentation] and the https://github.com/hazelcast/hazelcast-csharp-client/tree/master/src/Hazelcast.Net.Examples[code samples]. 
```
Personally, I do think we need a Google override to allow use of i.e. and e.g. but also I really dislike the US English standard of using a comma after e.g.

* SQL Support: the client allows running SQL queries over distributed data.
* Near Cache: it supports Near Cache for faster read speeds. Eventual consistency is also supported.
* Security Features: the client offers SSL support and Mutual Authentication for enterprise-level security needs.
* JSON Object Support: it allows using and querying JSON objects.
* Zero Downtime Upgrades: Blue/Green failover for zero downtime during upgrades is supported.
* Scalability: the Hazelcast .NET Client is designed to scale up to hundreds of members and thousands of clients, making it suitable for large-scale applications.

These features make the Hazelcast .NET Client ideal for .NET applications requiring high-performance, scalable, and distributed data processing capabilities.

TIP: For the latest .NET API documentation, see http://hazelcast.github.io/hazelcast-csharp-client/{page-latest-supported-csharp-client}/api/index.html[Hazelcast .NET Client docs].

== Install the .NET client

This section explains how to download and install the Hazelcast .NET client using the published Hazelcast.Net package from NuGet. 

=== Requirements

> The Hazelcast .NET client is distributed as a NuGet package that targets .NET Standard versions 2.0 and 2.1, and specific .NET versions starting with 5.0. It can therefore be used in any application targetting .NET versions that support these .NET Standard versions:

Reviewer 2: Our latest version of the client is supporting .NET 6 and .NET 8 besides .NET Framework 4.6.2 and 4.8. We drop supporting when Microsoft drops the .NET version. 5 was dropped a while ago.

> * .NET Framework 4.6.2 and above, on Windows

Reviewer 2: 4.8 would be better. We have no direct support on 4.7.x

> * .NET Core, on Windows, Linux and MacOS

Author: done

           var options = new HazelcastOptionsBuilder().Build();
           await using var client = await HazelcastClientFactory.StartNewClientAsync(options);

           // Use the client (e.g., get a distributed map)
           var map = await client.GetMapAsync<string, string>("my-distributed-map");
           await map.SetAsync("key", "value");
           var value = await map.GetAsync("key");
           System.Console.WriteLine($"Value for 'key': {value}");

           // Dispose the client when done
           await client.DisposeAsync();
       }
   }
```
**3. Run your application:**

```sh
   dotnet run
```

This will connect your .NET application to a Hazelcast cluster and allow you to interact with distributed data structures like maps. For more advanced configuration (e.g., connecting to Hazelcast Cloud, using SSL, or customizing logging), refer to the https://hazelcast.github.io/hazelcast-csharp-client/latest/doc/getting-started.html#using-the-client[Getting Started documentation] and the https://github.com/hazelcast/hazelcast-csharp-client/tree/master/src/Hazelcast.Net.Examples[code samples]. 

> NOTE: Always dispose of the client when finished to properly close connections and release resources.

Reviewer 2: I think it can stay. It shows the proper way to use it. Also, we can mention that you can use the client concurrently instead of creating on demand for each request.
